hello-world/hello-world.yaml

```diff
@@ -0,0 +1,32 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  creationTimestamp: null
+  name: echo
+spec:
+  params:
+  - description: Text to echo
+    name: text
+    type: string
+  steps:
+  - args:
+    - $(params.text)
+    command:
+    - echo
+    image: alpine
+    name: echo
+    resources: {}
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  creationTimestamp: null
+  name: hello-world
+spec:
+  tasks:
+  - name: echo-hello-world
+    params:
+    - name: text
+      value: Hello, World!
+    taskRef:
+      bundle: quay.io/hacbs-release/hello-world:0.3-alpine
```

Comment:
sorry. there is something I do not understand. from this line I read that this taskRef essentially refers to itself?

my understand is that this will eventually cause a bundle to be built yet it will refer to itself?

Reply:
That was necessary in M3 as both the pipeline and the task are in the same bundle. Not sure if there’s a proper way to do this. This is also partly why I preferred bundles to contain tasks exclusively. It would remove this references.